### ISSUE TITLE
Key Problems?

### ISSUE THREAD

**Issue by real-intruder** [andrei-tatar/node-red-contrib-nora#77]
I have to change my key each day since the last update to 0.0.34.
Google says that my devices are not reachable.
Then i go to https://node-red-google-home.herokuapp.com/ and make a new Key.
with this key i go to Node red Konfiguration of Nora and then it Works, but only for a Day.

**Comment by samubd** [comment]
Same here. Glad to help if needed.

**Comment by andrei-tatar** [comment]
Any errors in node-red console?

**Comment by samubd** [comment]
it seems that someting's wrong with the storage of the offline token on server side. Every day I have to login, the heroku app generates a new token as if the previous one were lost. After that I have to update the configuration node in nodered and then redo the account linking in google home.
After the procedure everything works fine but after some time (a day more or less) I have to repeat the procedure...
No errors are displayed in the nodered console: all the nora node appear as "connected"

**Comment by Barabba11** [comment]
Maybe you have  problems in your phone? Because here all is working fine

**Comment by bertreb** [comment]
I don't have problems with the server, i still have the same token from the beginning.
And logging in to heroku doesn't generate a new token. It just generates a new string, but the unlaying token is still te same (unless you revoke the token).

**Comment by real-intruder** [comment]
It Works too, to restart my Raspberry. Then it Works with the old token/key.
also there are 2 temporary posibilitys:
1: Get new Token
2: Restart the Node Red Server
But this works only for a Day.
I have the Problem since my last Palette Update in Node Red. The Time bevore i had no Problems...

**Comment by real-intruder** [comment]
The token on Phone and on PC are different! But it works when there are devices listened.
When i log in i get every time a new Token.

The last 4 Chars in Node Red are "okRw"

With my today login i have a Token with "eBU4"

but it works after Server Restart:

Do you like NORA and find it useful? Consider donating Paypal Me
Your devices/state
{
  "Zuhause": {
    "devices": {
      "234aae0.c79be52": {
        "type": "light",
        "brightnessControl": true,
        "turnOnWhenBrightnessChanges": true,
        "colorControl": false,
        "name": "Sofa",
        "roomHint": "Wohnzimmer",
        "state": {
          "online": true,
          "on": false,
          "brightness": 100
        }
      },
      "b6c49c13.dde31": {
        "type": "switch",
        "name": "Verstärker",
        "roomHint": "Wohnzimmer",

and so on

**Comment by real-intruder** [comment]
![NORA](https://user-images.githubusercontent.com/66788243/84599693-35c89a80-ae74-11ea-94d5-91f338e5ac4a.jpg)

Now the Devices are Offline in Google Home again

**Comment by real-intruder** [comment]
Restart Flow Works too, to bring the devices Online again

**Comment by samubd** [comment]
> I don't have problems with the nora-server, i still have the same token from the beginning.
> And logging in to heroku doesn't generate a new token. It just generates a new string, but the unlaying token is still te same (unless you revoke the token).

no, in my case a new login into heroku app generates a new token thus the list of devices (the json under "revoke token") is empty...

**Comment by bertreb** [comment]
This is an explanation i got from @andrei-tatar 'The signed part of the token changes. It's because it contains some salt to make it more secure. You can read more about it here: https://jwt.io/. It will change the string value on each login but it's the same content.'

**Comment by real-intruder** [comment]
![Node Red](https://user-images.githubusercontent.com/66788243/84914676-29368300-b0bc-11ea-8d71-feab6b07a090.jpg)

All Devices Show Connected, but they are Offline!

![WhatsApp Image 2020-06-17 at 16 54 14](https://user-images.githubusercontent.com/66788243/84914792-53884080-b0bc-11ea-92ba-58a5d899463b.jpeg)

all i do is "aktualisieren" on the Tab of Nora Konfig

![aktualisieren](https://user-images.githubusercontent.com/66788243/84914925-79ade080-b0bc-11ea-97a0-a7780b01978d.jpg)

without any Change of the Token
and Deploy 

Then it Works again vor 1-2 Days or only some Hours.

Node Red v1.0.6
node-red-contrib-nora 0.0.34

PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=raspbian
ID_LIKE=debian

Linux raspberrypi 4.19.66-v7+ #1253 SMP Thu Aug 15 11:49:46 BST 2019 armv7l GNU/Linux

**Comment by real-intruder** [comment]
With which File of Node Red can i Help? Where are debug Files to Support the Developer?
Can i Run Something to Log the Error?

**Comment by real-intruder** [comment]
How can I install a previous version of Nora?

**Comment by real-intruder** [comment]
I have deinstalled Nora and reinstalled Nora 0.0.34 ... same Problem.
Now i habe deinstalled Nora and reinstalled Nora@0.0.31.

Let's see if the problem comes up again.

**Comment by real-intruder** [comment]
No Poblems since Yesterday with Version 0.0.31

**Comment by real-intruder** [comment]
Still no Problems with Version 0.0.31.
The Error must be in the Code of node-red-contrib-nora Plugin beetween 0.0.32-0.0.34

**Comment by samubd** [comment]
I have the very same issue.
What I notice is that every time I login into nora web application a new
token is issued and thus the device list is empty. As soon as I replace the
token with the new one everything restart working fine.

Il giorno mar 23 giu 2020 alle ore 10:31 real-intruder <
notifications@github.com> ha scritto:

**Comment by samubd** [comment]
Yesterday a new token have been issued after a login into the web application. I've updated the token into nodered configuration and everything was workng correctly.
A few minutes ago it stopped working: this is the console message that I see:
"nora (SWRXI): socket connection error: not authorized"
(using version 0.0.34)

**Comment by real-intruder** [comment]
Still have no Problems since i have reinstalled Version 0.0.31 since 14 days ago.
It must be a Problem in the Code of the Plugin nora. The other Components still the same.

**Comment by real-intruder** [comment]
> Yesterday a new token have been issued after a login into the web application. I've updated the token into nodered configuration and everything was workng correctly.

You dont´t need a new Key. When you restart the nora Plugin or the whole Server it works too.
The Problem is, that nora don´t connect to Nora Service and report the Devices.

The Key form a PC Login and a Smartphone Login can be difference, but the devices only shown, when nora Plugin is connected to the Nora Service...
